## 73% of end-end tests have passed
```
Total executed tests: 11
Failed tests: 3
Expected to fail tests: 0
Passed tests: 8
```
```
IDs of failed tests: 727230,703133,727229 
```
<h3>Failed tests (3)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=feat/swap-exchange-rate&PR_ID=22078&APK_NAME=22078.apk&TR_CASE_IDS=727230,703133,727229">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784984">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: Button </div><div># STEP: Check that removed user is not shown in the list anymore</div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:246: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: zQ3...dWXh5 was not restored as a contact from waku backup!
E    Device 1: zQ3...Vacac was not restored as a contact from waku backup!
E    Device 1: admin_open was not restored from waku-backup!!
E    Device 1: member_open was not restored from waku-backup!!
E    Device 1: admin_closed was not restored from waku-backup!!
E    Device 1: member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES1dgXra6hh6mRq2kY">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784986">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td>

```
Test setup failed: critical/test_wallet.py:36: in prepare_devices
    self.wallet_2.wallet_tab.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:83: in find_element
    raise exception
../views/base_element.py:76: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at runMicrotasks (<anonymous>)
E       at processTicksAndRejections (node:internal/process/task_queues:96:5)
E       at AndroidUiautomator2Driver.doFindElementOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/find.js:44:7)
E       at doFind (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:48:17)
E       at wrappedCondFn (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:71:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784985">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: WalletTab </div><div>Device 2: Find `WalletTab` by `accessibility id`: `wallet-stack-tab` </div></blockquote></p>

```
Test setup failed: critical/test_wallet.py:36: in prepare_devices
    self.wallet_2.wallet_tab.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:83: in find_element
    raise exception
../views/base_element.py:76: in find_element
    return self.driver.find_element(self.by, self.locator)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/webdriver.py:409: in find_element
    return self.execute(RemoteCommand.FIND_ELEMENT, {'using': by, 'value': value})['value']
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:345: in execute
    self.error_handler.check_response(response)
../../../../status-app-prs@tmp/venv/lib/python3.10/site-packages/appium/webdriver/errorhandler.py:122: in check_response
    raise exception_class(msg=message, stacktrace=format_stacktrace(stacktrace))
 An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E   Stacktrace:
E   UnknownError: An unknown server-side error occurred while processing the command. Original error: 'POST /element' cannot be proxied to UiAutomator2 server because the instrumentation process is not running (probably crashed). Check the server log and/or the logcat output for more details
E       at UIA2Proxy.command (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/jsonwp-proxy/proxy.js:355:13)
E       at runMicrotasks (<anonymous>)
E       at processTicksAndRejections (node:internal/process/task_queues:96:5)
E       at AndroidUiautomator2Driver.doFindElementOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/lib/commands/find.js:44:7)
E       at doFind (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:48:17)
E       at wrappedCondFn (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:137:14)
E       at spin (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:221:20)
E       at waitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/asyncbox/lib/asyncbox.js:238:10)
E       at AndroidUiautomator2Driver.implicitWaitForCondition (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/timeout.ts:139:12)
E       at AndroidUiautomator2Driver.findElOrEls (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/appium-android-driver/lib/commands/find.ts:71:5)
E       at AndroidUiautomator2Driver.findElOrElsWithProcessing (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:60:12)
E       at AndroidUiautomator2Driver.findElement (/root/lambda/shared-tools/volume/uiautomator/uiautomator-2.32.3/node_modules/@appium/base-driver/lib/basedriver/commands/find.ts:75:12)
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (8)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784983">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESPcYt8iQIWabv0cma">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESzsDvYZaTyYnal33u">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784981">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES1dgXra6hh6mRq2kY">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784989">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784991">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784988">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784990">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784987">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3784982">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESmMhtbbY5P6WdvJMG">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESZ7yvNOvSeSd5dwY6">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/swap/setup_swap/style.cljs
+++ b/src/status_im/contexts/wallet/swap/setup_swap/style.cljs
@@ -1,4 +1,5 @@
-(ns status-im.contexts.wallet.swap.setup-swap.style)
+(ns status-im.contexts.wallet.swap.setup-swap.style
+  (:require [quo.foundations.colors :as colors]))
 
 (def container
   {:flex 1})
@@ -40,3 +41,22 @@
   {:height     :auto
    :min-height 40
    :max-height 62})
+
+(defn exchange-rate-loader
+  [theme]
+  {:margin-top       16
+   :width            72
+   :height           14
+   :border-radius    6
+   :background-color (colors/theme-colors colors/neutral-5 colors/neutral-90 theme)})
+
+(def exchange-rate-container
+  {:margin-top     16
+   :flex-direction :row})
+
+(def exchange-rate-crypto-label
+  {:color colors/neutral-50})
+
+(defn exchange-rate-fiat-label
+  [theme]
+  {:color (colors/theme-colors colors/neutral-40 colors/neutral-60 theme)})

--- a/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
+++ b/src/status_im/contexts/wallet/swap/setup_swap/view.cljs
@@ -343,6 +343,33 @@
                           (= (:symbol token)
                              (:symbol asset-to-pay)))}]))
 
+(defn- swap-exchange-rate-view
+  []
+  (let [theme                     (quo.theme/use-theme)
+        asset-to-pay              (rf/sub [:wallet/swap-asset-to-pay])
+        asset-to-receive          (rf/sub [:wallet/swap-asset-to-receive])
+        swap-exchange-rate-crypto (rf/sub [:wallet/swap-exchange-rate-crypto])
+        swap-exchange-rate-fiat   (rf/sub [:wallet/swap-exchange-rate-fiat])
+        loading-swap-proposal?    (rf/sub [:wallet/swap-loading-swap-proposal?])]
+    (cond
+      loading-swap-proposal?
+      [rn/view {:style (style/exchange-rate-loader theme)}]
+      swap-exchange-rate-crypto
+      [rn/view {:style style/exchange-rate-container}
+       [quo/text
+        {:weight :medium
+         :size   :paragraph-2
+         :style  style/exchange-rate-crypto-label}
+        (i18n/label :t/swap-exchange-rate-in-crypto
+                    {:receive-token-symbol (:symbol asset-to-receive)
+                     :exchange-rate        swap-exchange-rate-crypto
+                     :pay-token-symbol     (:symbol asset-to-pay)})]
+       [quo/text
+        {:weight :medium
+         :size   :paragraph-2
+         :style  (style/exchange-rate-fiat-label theme)}
+        (str " (" swap-exchange-rate-fiat ")")]])))
+
 (defn view
   []
   (let [[pay-input-state set-pay-input-state]       (rn/use-state controlled-input/init-state)
@@ -495,7 +522,7 @@
        :params        {:show-account-balances? true
                        :asset-symbol           (:symbol asset-to-pay)
                        :network                network}}]
-     [rn/view {:style style/inputs-container}
+     [rn/scroll-view {:style style/inputs-container}
       [pay-token-input
        {:input-state      pay-input-state
         :on-max-press     on-max-press
@@ -514,7 +541,8 @@
       [receive-token-input
        {:input-focused? (not pay-input-focused?)
         :on-token-press #(rf/dispatch [:show-bottom-sheet {:content receive-token-bottom-sheet}])
-        :on-input-focus #(set-pay-input-focused? false)}]]
+        :on-input-focus #(set-pay-input-focused? false)}]
+      [swap-exchange-rate-view]]
      [rn/view {:style style/footer-container}
       (when-not loading-swap-proposal? [alert-banner {:pay-input-error? pay-input-error?}])
       (when (or loading-swap-proposal? swap-proposal)

--- a/src/status_im/subs/wallet/swap.cljs
+++ b/src/status_im/subs/wallet/swap.cljs
@@ -333,3 +333,43 @@
                           :token            token-for-fees
                           :prices-per-token prices-per-token})]
      fee-in-fiat)))
+
+(rf/reg-sub
+ :wallet/swap-exchange-rate-crypto
+ :<- [:wallet/swap-proposal-amount-in]
+ :<- [:wallet/swap-asset-to-pay]
+ :<- [:wallet/swap-proposal-amount-out]
+ :<- [:wallet/swap-asset-to-receive]
+ (fn [[amount-in asset-to-pay amount-out asset-to-receive]]
+   (let [pay-token-decimals     (:decimals asset-to-pay)
+         receive-token-decimals (:decimals asset-to-receive)
+         amount-in-num          (some-> amount-in
+                                        (number/hex->whole pay-token-decimals)
+                                        (number/to-fixed pay-token-decimals)
+                                        (money/->bignumber))
+         amount-out-num         (some-> amount-out
+                                        (number/hex->whole receive-token-decimals)
+                                        (number/to-fixed receive-token-decimals)
+                                        (money/->bignumber))
+         exchange-rate          (when (and amount-in-num amount-out-num)
+                                  (money/div amount-in-num amount-out-num))
+         display-decimals       (min pay-token-decimals
+                                     constants/min-token-decimals-to-display)]
+     (when exchange-rate
+       (number/to-fixed exchange-rate display-decimals)))))
+
+(rf/reg-sub
+ :wallet/swap-exchange-rate-fiat
+ :<- [:wallet/swap-exchange-rate-crypto]
+ :<- [:wallet/swap-asset-to-pay]
+ :<- [:profile/currency]
+ :<- [:profile/currency-symbol]
+ :<- [:wallet/prices-per-token]
+ (fn [[swap-exchange-rate asset-to-pay currency currency-symbol prices-per-token]]
+   (when swap-exchange-rate
+     (let [fiat-value (utils/calculate-token-fiat-value
+                       {:currency         currency
+                        :balance          swap-exchange-rate
+                        :token            asset-to-pay
+                        :prices-per-token prices-per-token})]
+       (utils/fiat-formatted-for-ui currency-symbol fiat-value)))))

--- a/translations/en.json
+++ b/translations/en.json
@@ -2566,6 +2566,7 @@
   "sun": "Sun",
   "swap": "Swap",
   "swap-details": "Swap details",
+  "swap-exchange-rate-in-crypto": "1 {{receive-token-symbol}} ≈ {{exchange-rate}} {{pay-token-symbol}}",
   "swap-failed": "Swap failed",
   "swapped-to": "Swapped {{pay-amount}} {{pay-token-symbol}} to {{receive-amount}} {{receive-token-symbol}}",
   "swapping-to": "Swapping {{pay-amount}} {{pay-token-symbol}} to {{receive-amount}} {{receive-token-symbol}}",

